### PR TITLE
Update immigration information on application details screen

### PR DESCRIPTION
**Pull request by simonwhatley**
See: [⛵ Playing back information about residency & immigration status to providers to help them better assess international candidates eligibility](https://trello.com/c/PmQ8alK3)

When entering personal information, candidates are asked the following questions:

```
1. What is your nationality?

[] British --> Go to Q2
[] Irish --> Go to Q2
[] Other
   Give details --> Go to Q3
   [                              ]
  [Add another nationality]

[Save and continue]


2. Will you have lived in the UK in the 3 years before your teacher training starts?

() yes --> Go to personal details summary
() no --> Go to personal details summary

[Save and continue]


3. Do you have the right to work or study in the UK for the length of the teacher training course?

() yes --> Go to Q4
() not yet --> Go to Q5

[Save and continue]


4. What is your immigration status?

() EU settled status --> Go to Q6
() EU pre-settled status --> Go to Q6
() other
  Give details --> Go to Q6
  [                              ]

[Save and continue]


5. How will you get the right to work or study in the UK?

() a visa sponsored by a course provider --> Go to personal details summary
() another route --> Go to personal details summary
  Give details
  [                              ]

[Save and continue]


6. When did you enter the UK?

Day Month Year --> Go to personal details summary
[  ] [  ] [    ]

[Save and continue]
```

We replay these on the application details screen where relevant.

### CODE REVIEW

File: app/data/generators/personal-details.js [hunk 1]
```diff
@@ -1,56 +1,151 @@
...
+
 const generatorHelpers = require('../helpers/generators')
 
 module.exports = () => {
+
+  // ---------------------------------------------------------------------------
+  // Nationality
+  // ---------------------------------------------------------------------------
   const nationalities = {
     british: ['British'],
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:14]: I think it's misleading to model a single value as an array

File: app/data/generators/personal-details.js [hunk 2]
```diff
@@ -1,56 +1,151 @@
...
+        visa: 0.7,
+        other: 0.3
+      })
+
+      rightToWorkStudyHow = rightToWorkStudyHowOptions[selectedRightToWorkStudyHow]
+
+      if (rightToWorkStudyHow === 'Another route') {
+
+        let rightToWorkStudyHowChoices = [
+          'I am applying for a permanent residence card',
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:96]: Contract

File: app/data/generators/personal-details.js [hunk 3]
```diff
@@ -1,56 +1,151 @@
...
+          'I am applying for a visa',
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:97]: Contract

File: app/data/generators/personal-details.js [hunk 4]
```diff
@@ -1,56 +1,151 @@
...
+          'I have refugee status',
+          'I’m applying for a student visa',
+          'I’m applying for a spousal visa'
+        ]
+
+        let rightToWorkStudyHowDetails = rightToWorkStudyHowChoices.sort(() => 0.5 - Math.random())
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:103]: What does this do - perhaps a comment is needed

File: app/data/generators/personal-details.js [hunk 5]
```diff
@@ -1,56 +1,151 @@
...
+  // ---------------------------------------------------------------------------
+  // Date entered UK
+  // ---------------------------------------------------------------------------
+  let dateEnteredUK
+
+  if (isInternationalCandidate) {
+
+    if (rightToWorkStudy === 'Yes') {
+      dateEnteredUK = faker.date.between('2000-01-01','2016-08-31')
+      dateEnteredUK = DateTime.fromJSDate(dateEnteredUK).toFormat('yyyy-LL-dd')
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:147]: Combine to one line?

Comment (adamsilver) [reply app/data/generators/personal-details.js:147]: How come this particular format?

File: app/data/generators/personal-details.js [hunk 6]
```diff
@@ -84,7 +179,8 @@ module.exports = () => {
     y65_over: 0
   })
 
-  const dateOfBirth = dateOfBirthOptions[selectedDateOfBirth]
+  let dateOfBirth = dateOfBirthOptions[selectedDateOfBirth]
+  dateOfBirth = DateTime.fromJSDate(dateOfBirth).toFormat('yyyy-LL-dd')
```
Comment (adamsilver) [inline app/data/generators/personal-details.js:183]: I'm worried about "fromJSDATE" being used.

File: app/data/generators/personal-details.js (hunk 3 above)
Comment (simonwhatley) [reply app/data/generators/personal-details.js:97]: Changed to `I'm`

File: app/data/generators/personal-details.js (hunk 4 above)
Comment (simonwhatley) [reply app/data/generators/personal-details.js:103]: Changed to `let rightToWorkStudyHowDetails = faker.helpers.randomize(rightToWorkStudyHowChoices)`

File: app/data/generators/personal-details.js (hunk 5 above)
Comment (simonwhatley) [reply app/data/generators/personal-details.js:147]: It's a valid date. Faker creates a time, which is unnecessary.

File: app/data/generators/personal-details.js (hunk 6 above)
Comment (simonwhatley) [reply app/data/generators/personal-details.js:185]: It's a valid date. Faker creates a time, which is unnecessary, so I've removed.

**Pull request merged**